discovery/kubernetes/pod.go

```diff
@@ -87,7 +109,12 @@ func (p *Pod) enqueue(obj interface{}) {
 func (p *Pod) Run(ctx context.Context, ch chan<- []*targetgroup.Group) {
 	defer p.queue.ShutDown()
 
-	if !cache.WaitForCacheSync(ctx.Done(), p.informer.HasSynced) {
+	cacheSyncs := []cache.InformerSynced{p.podInf.HasSynced}
+	if p.nodeInf != nil {
```

Comment:
I don't know if it's a good idea to use the value of the informer as a discriminator of whether node metadata should be attached or not. Any feedback is welcome.

Reply:
I think that deserves a comment at least, otherwise a config field on the pod role would be ok as well I think

Reply:
I like the field on the pod role 👍